data/pools/transactionPool.go

```diff
@@ -315,77 +306,23 @@ func (pool *TransactionPool) checkPendingQueueSize(txnGroup []transactions.Signe
 	return nil
 }
 
-// FeePerByte returns the current minimum microalgos per byte a transaction
-// needs to pay in order to get into the pool.
-func (pool *TransactionPool) FeePerByte() uint64 {
-	return pool.feePerByte.Load()
-}
-
-// computeFeePerByte computes and returns the current minimum microalgos per byte a transaction
-// needs to pay in order to get into the pool. It also updates the atomic counter that holds
-// the current fee per byte
-func (pool *TransactionPool) computeFeePerByte() uint64 {
-	// The baseline threshold fee per byte is 1, the smallest fee we can
-	// represent.  This amounts to a fee of 100 for a 100-byte txn, which
-	// is well below MinTxnFee (1000).  This means that, when the pool
-	// is not under load, the total MinFee dominates for small txns,
-	// but once the pool comes under load, the fee-per-byte will quickly
-	// come to dominate.
-	feePerByte := uint64(1)
-
-	// The threshold is multiplied by the feeThresholdMultiplier that
-	// tracks the load on the transaction pool over time.  If the pool
-	// is mostly idle, feeThresholdMultiplier will be 0, and all txns
-	// are accepted (assuming the BlockEvaluator approves them, which
-	// requires a flat MinTxnFee).
-	feePerByte = feePerByte * pool.feeThresholdMultiplier
-
-	// The feePerByte should be bumped to 1 to make the exponentially
-	// threshold growing valid.
-	if feePerByte == 0 && pool.numPendingWholeBlocks > 1 {
-		feePerByte = uint64(1)
-	}
-
-	// The threshold grows exponentially if there are multiple blocks
-	// pending in the pool.
-	// golang has no convenient integer exponentiation, so we just
-	// do this in a loop
-	for i := 0; i < int(pool.numPendingWholeBlocks)-1; i++ {
-		feePerByte *= pool.expFeeFactor
-	}
-
-	// Update the counter for fast reads
-	pool.feePerByte.Store(feePerByte)
-
-	return feePerByte
-}
-
-// checkSufficientFee take a set of signed transactions and verifies that each transaction has
-// sufficient fee to get into the transaction pool
+// checkSufficientFee take a set of signed transactions and verifies that each
+// transaction has sufficient fee to get into the transaction pool. If we reject
+// a transaction unjustly, the caller will be sad, but it is not a protocol
+// violation.  Similarly, if we allow a transaction in, the fee will be further
+// checked in the evaluation of the transaction.
 func (pool *TransactionPool) checkSufficientFee(txgroup []transactions.SignedTxn) error {
-	// Special case: the state proof transaction, if issued from the
-	// special state-proof-sender address, in a singleton group, pays
-	// no fee.
-	if len(txgroup) == 1 {
-		t := txgroup[0].Txn
-		if t.Type == protocol.StateProofTx && t.Sender == transactions.StateProofSender && t.Fee.IsZero() {
-			return nil
-		}
```

Comment:
This is now handled by having the stateproof transaction not contribute in `SummarizeTxnFees`.